src/arrayMethodReduce.js

```diff
@@ -5,7 +5,13 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let previousValue = initialValue !== undefined ? initialValue : this[0];
+
+    for (let i = initialValue !== undefined ? 0 : 1; i < this.length; i++) {
```

Comment:
Don't mix loop conditions and ternary operators. Check ternary prior to that and use variable.